### PR TITLE
Added Login and Register Page

### PR DESCRIPTION
**Pull request by KMSZamil**
I have updated two pages "Login" and "Register". Accessible from /login and /register.

image

image

### CODE REVIEW

**Review by skfarhad — CHANGES_REQUESTED** [review]
Could you change the color scheme so that it's consistent with this design?

**Comment by KMSZamil** [comment]
The color is turned to white.

image

Hopefully change the design as designed

**Comment by KMSZamil** [comment]
Added new login screen

image

Added new register screen

image

**Review by Muttakee31 — APPROVED** [review]
LGTM. The merge conflict seems to be a simple one.
Just two suggestions.

1. We can remove the Facebook logo if it is not being used for the moment.
2. We may integrate an icon library like font awesome instead of using SVG images. It depends on the Figma design. if there are a significant amount of icons, we can adopt that.

File: web_app/hazard-reporting-map/components/LoginForm.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    const onSubmit = async (event) => {
+        event.preventDefault();
+        try {
+            const formData = new FormData(event.currentTarget);
+            const response = await login(formData);
+            if (!!response.error) {
+                setError(response.error.message);
+            } else {
+                //router.push("/");
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/components/LoginForm.jsx:25]: remove comments

File: web_app/hazard-reporting-map/components/LoginForm.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            }
+        } catch (error) {
+            //setError(error.message);
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/components/LoginForm.jsx:28]: remove comment

File: web_app/hazard-reporting-map/components/RegistrationForm.jsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        } catch (error) {
+            setError(error.message);
+        }
+    };
+
+    return (
+        <>
+            {error && <div className="text-2xl text-center text-red-500">{error}</div>}
+            <form className="login-form" onSubmit={onSubmit}>
+                <div>
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/components/RegistrationForm.jsx:43]: use mapping array for repeated elements

File: web_app/hazard-reporting-map/package.json [hunk 4]
```diff
@@ -9,10 +9,12 @@
     "lint": "next lint"
   },
   "dependencies": {
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/package.json:11]: you havent use these two packages . please remove it if its not necessary

**Comment by Shabab007** [comment]
Design feedbacks according to figma:
1. header title "System" should be in next line and in center
2. place holders are not given
3. Fonts colors are not matching 
4. no picture in right side
5. gap between the login and google sign in button not correct 
6. i agree not aligned . highlighted texts not underline
7. on register page thana, org etc input dropdowns and placeholders not added

**Review by Shabab007 — CHANGES_REQUESTED** [review]

File: web_app/hazard-reporting-map/package.json (hunk 4 above)
Comment (KMSZamil) [reply web_app/hazard-reporting-map/package.json:11]: I have removed unused packages

**Comment by KMSZamil** [comment]
Resolved as said

File: web_app/hazard-reporting-map/components/RegistrationForm.jsx [hunk 5]
```diff
@@ -41,33 +41,49 @@ const RegistrationForm = () => {
             {error && <div className="text-2xl text-center text-red-500">{error}</div>}
             <form className="login-form" onSubmit={onSubmit}>
                 <div>
-                    <label htmlFor="fname">Full Name</label>
-                    <input type="text" name="fname" id="fname" />
+                    <label htmlFor="fname" className="text-gray-600">
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/components/RegistrationForm.jsx:44]: only Fullname , mobile and email field are in the register form now according to update

File: web_app/hazard-reporting-map/components/RegistrationForm.jsx [hunk 6]
```diff
@@ -41,33 +41,49 @@ const RegistrationForm = () => {
...
+                    </label>
+                    <input type="text" name="fname" id="fname" placeholder="Enter your name" />
                 </div>
 
                 <div>
-                    <label htmlFor="mobile">Mobile No.</label>
-                    <input type="text" name="mobile" id="mobile" />
+                    <label htmlFor="mobile" className="text-gray-600">
+                        Mobile No.
+                    </label>
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/components/RegistrationForm.jsx:53]: add email field and validation also

File: web_app/hazard-reporting-map/components/SocialLogin.jsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
+"use client";
+
+import Image from "next/image";
+import Link from "next/link";
+
+const SocialLogins = ({ mode }) => {
+    const handleAuth = (e) => {
+        if (e === "google") {
+            signIn("google", { callbackUrl: "http://localhost:3000/" });
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/components/SocialLogin.jsx:9]: try not to use hard coded values use env for callbackUrl or use variable instead

File: web_app/hazard-reporting-map/styles/globals.css [hunk 8]
```diff
@@ -2,36 +2,96 @@
 @tailwind components;
 @tailwind utilities;
 
-:root {
+/* :root {
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/styles/globals.css:5]: remove comments

**Review by Shabab007 — CHANGES_REQUESTED** [review]

**Comment by Shabab007** [comment]
Lastly , Resolve conflicts

File: web_app/hazard-reporting-map/components/RegistrationForm.jsx [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+                <div>
+                    <label htmlFor="mobile" className="text-gray-600">
+                        Mobile No.
+                    </label>
+                    <input
+                        type="text"
+                        name="mobile"
+                        id="mobile"
+                        placeholder="Enter your mobile number"
+                        pattern="\d{11}"
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/components/RegistrationForm.jsx:57]: need to add mobile number validation

File: web_app/hazard-reporting-map/components/RegistrationForm.jsx [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+                        title="Please enter a valid 10-digit mobile number"
+                        required
+                    />
+                </div>
+
+                <div>
+                    <label htmlFor="email" className="text-gray-600">
+                        Email
+                    </label>
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/components/RegistrationForm.jsx:66]: email validation is missing

**Review by Shabab007 — CHANGES_REQUESTED** [review]

**Comment by Shabab007** [comment]
Please remove the old pics and upload updated pics or video so that we can look at the updated design

**Comment by KMSZamil** [comment]
Updated in the first comment @Shabab007 bhai

**Comment by Shabab007** [comment]
@KMSZamil  design looks good . 
1. only top margin or space of the title not matching . in figma there is much space on the top compared to your pic 
2. Please add email and mobile number validation for register input. if you need help let me know

**Comment by KMSZamil** [comment]
@Shabab007 bhai, Updated as requested

image

image

**Review by Shabab007 — APPROVED** [review]

File: web_app/hazard-reporting-map/styles/globals.css [hunk 11]
```diff
@@ -4,6 +4,7 @@
 
 
 body {
+  font-family: 'Manrope', sans-serif;
```
Comment (Shabab007) [inline web_app/hazard-reporting-map/styles/globals.css:40]: there is a way to incorporate font in nextjs and tailwind. please correct that

**Review by Shabab007 — CHANGES_REQUESTED** [review]

**Review by Shabab007 — APPROVED** [review]

**Review by skfarhad — APPROVED** [review]
Looks good!

**Pull request merged**